### PR TITLE
HDDS-7088. Removed IOException and added log level warn

### PR DESCRIPTION
**Pull request by navinko**
## What changes were proposed in this pull request?

To fix the issue once OM and SCM are colocated and OM fails to start we are just logging the warn messages instead of throwing IOException.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-7088

## How was this patch tested?

(Please explain how this patch was tested. Ex: unit tests, manual tests)
(If this patch involves UI changes, please attach a screen-shot; otherwise, remove this)

### CODE REVIEW

**Comment by navinko** [comment]
Hi @szetszwo added the suggested changes.  Kindly review the PR 
Thanks

**Comment by myskov** [comment]
Hi @navinko. As far as I understand the jira, the issue is not that OM doesn't start but the raft directory gets overridden (by OM or SCM, dependent on whom started later). In this case, OM should not be started.
It seems that the fix to the issue should allow sharing ratis dir by OM and SCM.

**Review by myskov — COMMENTED** [review]
Hi @navinko. As far as I understand the jira, the issue is not that OM doesn't start but the raft directory gets overridden (by OM or SCM, dependent on whom started later). In this case, OM should not be started.
It seems that the fix to the issue should allow sharing ratis dir by OM and SCM.

**Comment by szetszwo** [comment]
> ... raft directory gets overridden (by OM or SCM, dependent on whom started later) ...

@myskov , an existing raft directory should never be overridden automatically since it may cause data loss.  Currently, it throws an exception and stop starting.  After this change, it will print a warning and use a different directory because the group ids are different.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

Will wait for the build to be completed.

**Comment by navinko** [comment]
Thanks @szetszwo @myskov for suggestions and reviewing .

**Comment by myskov** [comment]
> > ... raft directory gets overridden (by OM or SCM, dependent on whom started later) ...
> 
> @myskov , an existing raft directory should never be overridden automatically since it may cause data loss. Currently, it throws an exception and stop starting. After this change, it will print a warning and use a different directory because the group ids are different.

Ok, but in this case, the warning message is misleading, and it should be changed.

**Review by myskov — CHANGES_REQUESTED** [review]

**Comment by szetszwo** [comment]
@myskov , you are right that the message is misleading.

@navinko , could you update the error message?

**Comment by myskov** [comment]
Let me explain the reasoning behind the requested change. We use Ozone in production and our administrators would be very concerned by the warning: "Ratis group Dir on disk scm does not match with RaftGroupID7f6848f2-63c2-3ce4-a1d2-9e88065dfcc3 generated from service id ozone1. Looks like there is a change to ozone.om.service.ids value after the cluster is setup. Currently change to this value is not supported."
It sounds like a fatal error, not a warning.

**Comment by navinko** [comment]
Thanks @szetszwo @myskov for reviewing . I have updated the PR

**Comment by kerneltime** [comment]
@navinko why should the exception be dropped? An incorrectly configured OM and SCM folder should be a catastrophic failure.

**Comment by navinko** [comment]
Hi @kerneltime , In my understanding it's very common in production that SCM and OM wud be collocated and in current design if both the scm and om ratis directory shared and if which ever is started later , it will not just override the existing Raft directory as there will be data loss so it throws an exception and stop starting. After this change, it will print a warning and  create a new directory because the group ids are different.

**Comment by szetszwo** [comment]
> ... In my understanding it's very common in production that SCM and OM wud be collocated ...

Indeed, it is uncommon in production (SCM and OM usually use different directories, or even different machine) but it may happen in testing setup.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java [hunk 1]
```diff
@@ -1351,13 +1351,10 @@ private void initializeRatisDirs(OzoneConfiguration conf) throws IOException {
...
           if (ratisGroupDir.isDirectory()) {
             if (!ratisGroupDir.getName().equals(groupIDfromServiceID)) {
-              throw new IOException("Ratis group Dir on disk "
-                  + ratisGroupDir.getName() + " does not match with RaftGroupID"
-                  + groupIDfromServiceID + " generated from service id "
-                  + getOMServiceId() + ". Looks like there is a change to " +
-                  OMConfigKeys.OZONE_OM_SERVICE_IDS_KEY + " value after the " +
-                  "cluster is setup. Currently change to this value is not " +
-                  "supported.");
+              LOG.warn("Another ratis dir is detected "
```
Comment (szetszwo) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java:1354]: @navinko , Please include the name `ratisGroupDir.getName()`.

Comment (szetszwo) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java:1354]: @navinko , thanks for the update.  Let's use a similar message for both cases
```java
@@ -1353,16 +1353,13 @@ private void initializeRatisDirs(OzoneConfiguration conf) throws IOException {
         for (File ratisGroupDir : ratisDirFiles) {
           if (ratisGroupDir.isDirectory()) {
             if (!ratisGroupDir.getName().equals(groupIDfromServiceID)) {
-              throw new IOException("Ratis group Dir on disk "
-                  + ratisGroupDir.getName() + " does not match with RaftGroupID"
-                  + groupIDfromServiceID + " generated from service id "
-                  + getOMServiceId() + ". Looks like there is a change to " +
-                  OMConfigKeys.OZONE_OM_SERVICE_IDS_KEY + " value after the " +
-                  "cluster is setup. Currently change to this value is not " +
-                  "supported.");
+              LOG.warn("Unknown directory {} exists in ratis storage dir {}."
+                  + " It is recommended not to share the ratis storage dir.",
+                  ratisGroupDir, omRatisDir);
             }
           } else {
-            LOG.warn("Unknown file {} exists in ratis storage dir {}",
+            LOG.warn("Unknown file {} exists in ratis storage dir {}. "
+                + " It is recommended not to share the ratis storage dir.",
                 ratisGroupDir, omRatisDir);
           }
         }
```

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by navinko** [comment]
Thanks @szetszwo @myskov @kerneltime for reviewing the PR.

**Pull request merged**